loader/icd.c

```diff
@@ -79,6 +79,17 @@ void khrIcdVendorAdd(const char *libraryName)
         goto Done;
     }
 
+    // get the library's file name
+    const char *libName = libraryName;
+    const char *c;
+    for (c = libraryName; *c; ++c)
+    {
+        if (*c == DIRECTORY_SYMBOL)
+        {
+            libName = c + 1;
+        }
+    }
```

Comment:
This may be a non-issue, but so we go in with eyes open: As I understand this change strips all path information from the library name and only loads the first library with a given name.  While this will prevent unintentional scenarios where the same library name would be loaded multiple times, it also means that library names must be unique, even if they are in different directories, or the ICD loader will filter them out.  Is this the behavior we want?

I'm not aware of any implementors doing this, but a scenario I've been thinking of where two libraries could legitimately have the same name is if one device generation is supported by one software package and installed into one directory and another device generation is supported by a different package and installed into a different directory, but both device generations have the same library file name.

Reply:
Yes. This is the behavior we want. 
See my previous comment. ICD's rule on duplicate libraries has always been like this. 

There is no such scenario at least for us. (Support for multiple device generations is not implemented in this way.) 
Unlike loading OpenCL libraries form different vendors, loading multiple libraries from the same vendor is undefined behavior.